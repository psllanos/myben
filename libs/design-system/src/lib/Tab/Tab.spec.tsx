import { fireEvent, render, screen } from '@testing-library/react'
import { Tab } from './'

// Not tested too thoroughly, most logic is covered by Headless UI base
describe('Tab', () => {
    it('should render correctly', () => {
        const component = render(
            <Tab.Group>
                <Tab.List>
                    <Tab>Tab 1</Tab>
                    <Tab>Tab 2</Tab>
                </Tab.List>
                <Tab.Panels>
                    <Tab.Panel>Content 1</Tab.Panel>
                    <Tab.Panel>Content 2</Tab.Panel>
                </Tab.Panels>
            </Tab.Group>
        )

        expect(screen.getByText('Tab 1')).toBeInTheDocument()
        expect(screen.queryByText('Content 1')).toBeInTheDocument()
        expect(component).toMatchSnapshot()
    })

    describe('when toggled', () => {
        it('should render the correct panel', () => {
            render(
                <Tab.Group>
                    <Tab.List>
                        <Tab>Tab 1</Tab>
                        <Tab>Tab 2</Tab>
                    </Tab.List>
                    <Tab.Panels>
                        <Tab.Panel>Content 1</Tab.Panel>
                        <Tab.Panel>Content 2</Tab.Panel>
                    </Tab.Panels>
                </Tab.Group>
            )

            // Content 1 visible, Content 2 hidden
            expect(screen.getByText('Content 1')).toBeInTheDocument()
            expect(screen.queryByText('Content 2')).not.toBeInTheDocument()

            fireEvent.click(screen.getByText('Tab 2'))

            // Content 2 visible, content 1 hidden
            expect(screen.getByText('Content 2')).toBeInTheDocument()
            expect(screen.queryByText('Content 1')).not.toBeInTheDocument()
        })
    })
})
