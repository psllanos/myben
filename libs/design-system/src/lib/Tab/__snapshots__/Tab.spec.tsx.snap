// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-orientation="horizontal"
        class="inline-flex items-center space-x-1 p-1 rounded-lg bg-gray-700 text-white"
        role="tablist"
      >
        <button
          aria-controls="headlessui-tabs-panel-:r2:"
          aria-selected="true"
          class="grow flex items-center justify-center text-base py-1 px-4 rounded focus:outline-none focus:ring-2 focus:ring-opacity-60 focus:ring-gray-400 bg-gray-400 hover:bg-gray-300 text-white shadow"
          data-headlessui-state="selected"
          id="headlessui-tabs-tab-:r0:"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab 1
        </button>
        <button
          aria-controls="headlessui-tabs-panel-:r3:"
          aria-selected="false"
          class="grow flex items-center justify-center text-base py-1 px-4 rounded focus:outline-none focus:ring-2 focus:ring-opacity-60 focus:ring-gray-400 hover:bg-gray-600 text-gray-100"
          data-headlessui-state=""
          id="headlessui-tabs-tab-:r1:"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab 2
        </button>
      </div>
      <div
        class=""
      >
        <div
          aria-labelledby="headlessui-tabs-tab-:r0:"
          class="focus:outline-none"
          data-headlessui-state="selected"
          id="headlessui-tabs-panel-:r2:"
          role="tabpanel"
          tabindex="0"
        >
          Content 1
        </div>
        <span
          aria-labelledby="headlessui-tabs-tab-:r1:"
          id="headlessui-tabs-panel-:r3:"
          role="tabpanel"
          style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
          tabindex="-1"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-orientation="horizontal"
      class="inline-flex items-center space-x-1 p-1 rounded-lg bg-gray-700 text-white"
      role="tablist"
    >
      <button
        aria-controls="headlessui-tabs-panel-:r2:"
        aria-selected="true"
        class="grow flex items-center justify-center text-base py-1 px-4 rounded focus:outline-none focus:ring-2 focus:ring-opacity-60 focus:ring-gray-400 bg-gray-400 hover:bg-gray-300 text-white shadow"
        data-headlessui-state="selected"
        id="headlessui-tabs-tab-:r0:"
        role="tab"
        tabindex="0"
        type="button"
      >
        Tab 1
      </button>
      <button
        aria-controls="headlessui-tabs-panel-:r3:"
        aria-selected="false"
        class="grow flex items-center justify-center text-base py-1 px-4 rounded focus:outline-none focus:ring-2 focus:ring-opacity-60 focus:ring-gray-400 hover:bg-gray-600 text-gray-100"
        data-headlessui-state=""
        id="headlessui-tabs-tab-:r1:"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 2
      </button>
    </div>
    <div
      class=""
    >
      <div
        aria-labelledby="headlessui-tabs-tab-:r0:"
        class="focus:outline-none"
        data-headlessui-state="selected"
        id="headlessui-tabs-panel-:r2:"
        role="tabpanel"
        tabindex="0"
      >
        Content 1
      </div>
      <span
        aria-labelledby="headlessui-tabs-tab-:r1:"
        id="headlessui-tabs-panel-:r3:"
        role="tabpanel"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        tabindex="-1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
